core/src/main/java/org/infinispan/conflict/impl/DefaultConflictManager.java

```diff
@@ -470,5 +507,11 @@ public boolean tryAdvance(Consumer<? super Map<Address, CacheEntry<K, V>>> actio
          action.accept(iterator.next());
          return true;
       }
+
+      void stop() {
+         if (trace) log.tracef("Stop called on ReplicaSpliterator. Current segment %s", nextSegment);
+         if (isConflictResolutionInProgress())
```

Comment:
I think `stop()` could be called before any `tryAdvance()`, so I'd replace the condition with a `null` check on `segmentRequestFuture`.